### PR TITLE
Bug 1806838 - Update extension icon when invalidate is called.

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1806838

### CODE REVIEW

File: android-components/components/browser/menu/src/main/java/mozilla/components/browser/menu/item/WebExtensionBrowserMenuItem.kt [hunk 1]
```diff
@@ -51,40 +51,52 @@ class WebExtensionBrowserMenuItem(
...
+            menu.dismiss()
+        }
+    }
+
+    override fun invalidate(view: View) {
+        val labelView = view.findViewById<TextView>(R.id.action_label)
+        val badgeView = view.findViewById<TextView>(R.id.badge_text)
+        val imageView = view.findViewById<ImageView>(R.id.action_image)
+
+        updateItem(view)
```
Comment (Amejia481) [inline android-components/components/browser/menu/src/main/java/mozilla/components/browser/menu/item/WebExtensionBrowserMenuItem.kt:67]: @willdurand I think I found the source of the [bug 1806838](https://bugzilla.mozilla.org/show_bug.cgi?id=18068380, in invalidate we where updating the `label` and `badge` but no the icon, that was causing the previous icon was kept when invalidate is called.

Comment (Amejia481) [reply android-components/components/browser/menu/src/main/java/mozilla/components/browser/menu/item/WebExtensionBrowserMenuItem.kt:67]: Every time we get a new [browser or page action ](https://github.com/mozilla-mobile/firefox-android/blob/35ce01367157440f9e9daa4ed48a8022af80c8f2/android-components/components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenu.kt#L42-L43), we update the 
the [browserMenuItem.action](https://github.com/mozilla-mobile/firefox-android/blob/35ce01367157440f9e9daa4ed48a8022af80c8f2/android-components/components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenu.kt#L135) which contains all the data related to the browser or page action, and finally we call [invalidate](https://github.com/mozilla-mobile/firefox-android/blob/35ce01367157440f9e9daa4ed48a8022af80c8f2/android-components/components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenu.kt#L43) to update the data, I think in some cases, we were reusing the UI items and as the icon was not updated on invalidate, we were keeping the previous icon.

Comment (Amejia481) [reply android-components/components/browser/menu/src/main/java/mozilla/components/browser/menu/item/WebExtensionBrowserMenuItem.kt:67]: I was able to reproduce the issue by [following this report's steps](https://bugzilla.mozilla.org/show_bug.cgi?id=1806838#c27), I installed all the addons the reported had and opened the add-ons menu, put Fenix in background wait for about a minute and then opened it again, and I was able to see icon duplication.


image

**Review by willdurand — APPROVED** [review]
This looks good. I'd be good to have some test coverage if possible, though. Thanks!

**Comment by Amejia481** [comment]
Thanks for the review!
I'll try to find a way to improve the coverage!

**Comment by Amejia481** [comment]
Tests added :)

**Review by willdurand — APPROVED** [review]

**Pull request merged**